torch/utils/checkpoint.py

```diff
@@ -290,6 +293,8 @@ def checkpoint(
     *args,
     use_reentrant: Optional[bool] = None,
     context_fn: Callable[[], Tuple[ContextManager, ContextManager]] = noop_context_fn,
+    determinism_check: str = "default",
+    debug: bool = False,
```

Comment:
Why two different options? Why not put the determism check in the debug config?

Reply:
We want to separate them because determinism_check="default" is something the user should keep on by default.
but debug=True (running logging_mode all the time) can impact perf.

Reply:
But you could have `debug="default"` and "full" and "none" for the other options? Or do you think that there are cases where people will want debug without determinism checks?

Reply:
> Or do you think that there are cases where people will want debug without determinism checks?

Ahh no, I'd say the only concern is if we wish to add more types of determinism_checks in the future, for example to check tensor statistics, it would be a little harder to do if the two options are coupled.

Reply:
We could also have a `debug="full_values"` ? The same way you can add things to determinism_check?

Reply:
Right, I think it makes sense for "full_values" to always have verbose logs. 

But a trickier example is if we want to provide the option for the user to specify their own determinism_check. Someone may have a check that may or may not be intended to be on all the time.

A possible UX today is to allow the user to pass in a function to `determinism_check`, but that may be more difficult if we only had the `debug` option.